src/invoice.rs

```diff
@@ -160,12 +164,18 @@ impl FromStr for RgbInvoice {
 
         let mut next_path_index = 0;
 
-        let contract = match ContractId::from_str(&path[next_path_index]) {
+        let contract_id_str = &path[next_path_index];
+        let contract = match ContractId::from_str(contract_id_str) {
             Ok(cid) => {
                 next_path_index += 1;
                 Some(cid)
             }
-            Err(_e) => None,
+            Err(_e) => {
+                if !uri.path().to_string().starts_with('/') {
```

Comment:
not sure I get this part. Why do we need to check that the path doesn't start with `/`?

The rule that the first contract component must always be contract id. If it is not, this is `InvalidContractId` error. I think this `if` condition should be entirely omitted and we need just to error in this case.

Reply:
This check is there to distinguish from a missing contract ID and an invalid one. As I explained in https://github.com/RGB-WG/rgb-wallet/pull/33#discussion_r1167467937 path segments don't include empty segments, that's the reason I introduced this check. Tests should confirm code is now working as expected

Reply:
Oh, github didn't show me your comment originally...

Ok, I see. However we need to structure this check differently in order to avoid unnecessary allocation. Proposing the change in my new review